wordpress-mobile/WordPress-Android
# Blaze: Campaign Detail Page: introduce campaign id into navigation

Parent #18532

This PR introduces the passing of "campaignId" from card tap to CampaignDetailFragment
Re:  `campaign.campaignID.toInt()` -  this will be updated as part of the FluxC updates for stats response.

**Note**: To keep the PR reviews small and manageable, I'll be breaking the CampaignDetail into separate PRs. This PR does not contain any logic to show the actual campaign detail view

**To test:**
- Install and launch the app
- Select a site that has blaze campaigns
- Navigate to My Site
- Tap on the Campaign Detail within the Blaze Card
- ✅ Verify the Campaign Detail view is launched


## Regression Notes
1. Potential unintended areas of impact
The campaign detail view doesn't launch

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test

3. What automated tests I added (or what prevented me from doing so)
Updated `BlazeCardViewModelSliceTest` and `BlazeCardBuilderTest`

PR submission checklist:

UI Changes testing checklist: N/A
